src/test/groovy/projet/gl51/store/MemoryProductStorageSpec.groovy

```diff
@@ -0,0 +1,108 @@
+package projet.gl51.store
+
+import spock.lang.Specification
+
+class MemoryProductStorageSpec extends Specification {
+
+    ProductStorage storage = new MemoryProductStorage()
+
+
+	void "empty storage returns empty list"() {
+		expect:
+		storage.all() == []
+	}
+	
+    void "adding a product returns the product in the list with a new id"() {
+		setup:
+		Integer beforeSize = storage.all().size()
+		Product beforeProduct = new Product(name: name, description: description, price: price, idealTemperature: idealTemperature)
+		String productId = storage.save(beforeProduct)
+		
+		when:
+		List<Product> products = storage.all()
+		List<Product> afterProducts = products.findAll { it.id == productId }
+		
+		then:
+		products.size() == beforeSize + 1
+		afterProducts.size() == 1
+		beforeProduct.equals(afterProducts[0])
+		
+		where:
+		name | description | price | idealTemperature
+		"iphone" | "smartphone" | 1500 | 25 
+		"thunder" | "trotinette" | 5000 | 5 
+    }
+	
+	void "deleting a product removes it from the list"() {
+		setup:
+		Integer beforeSize = storage.all().size()
+		String productId = storage.save(new Product(name: name, description: description, price: price, idealTemperature: idealTemperature))
+		
+		when:
+		storage.delete(productId)
+		List<Product> products = storage.all()
+		Product afterProduct = products.find { it.id == productId }
+		
+		then:
+		products.size() == beforeSize
+		afterProduct == null
+		
+		where:
+		name | description | price | idealTemperature
+		"iphone" | "smartphone" | 1500 | 25
+		"thunder" | "trotinette" | 5000 | 5
+	}
+	
+	void "modifying a product changes it in the list"() {
+		setup:
+		String productId = storage.save(new Product(name: name, description: description, price: price, idealTemperature: idealTemperature))
+		
+		when:
+		Product newProduct = new Product(name: name1, description: description1, price: price1, idealTemperature: idealTemperature1)
+		storage.update(productId, newProduct)
+		Product afterProduct = storage.all().find { it.id == productId }
+		
+		then:
+		newProduct.equals(afterProduct)
```

Comment:
```suggestion
		newProduct == afterProduct
```

Reply:
https://objectpartners.com/2018/11/07/equals-is-compareto-and-the-groovy-identity-operator/